sprox/providerselector.py

```diff
@@ -111,14 +112,10 @@ def _get_engine(self, hint, hints):
 
     def get_entity(self, identifier, hint=None, **hints):
         engine = self._get_engine(hint, hints)
-        for mapper in _mapper_registry:
-            if mapper.class_.__name__ == identifier:
-                if engine is None:
-                    return mapper.class_
-                if engine is not None and mapper.tables[0].bind == engine:
-                    return mapper.class_
-
-        raise KeyError('could not find model by the name %s in %s'%(model_name, metadata))
+        try:
+            return mapped_class(engine, identifier)
+        except KeyError as ex:
+            raise NameError(str(ex))
```

Comment:
Why the behaviour change? Previously this was throwing a `KeyError('could not find model by the name %s in %s'%(model_name, metadata))`, now we throw a `NameError` with what message? What about existing codebase already catching the `KeyError`?

Reply:
It's not a change, there was a test, in test_providerselector that expected a NameError to be raised
```python
    @raises(NameError)
    def test_entity_not_found(self):
        self.selector.get_entity('Dummy')
```

Reply:
Interesting, how did that test now fail previously?